### PR TITLE
feat: introducing tracks for sessions

### PR DESCRIPTION
**Pull request by maze-runnar**
Screenshot from 2020-08-18 23-41-44
Screenshot from 2020-08-18 23-41-41
Screenshot from 2020-08-18 23-41-33

Adding more than 1 speaker for a session with track - 
Screenshot from 2020-08-18 23-57-54
Screenshot from 2020-08-18 23-57-50

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by maze-runnar** [comment]
@iamareebjamal please review.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/controllers/public/sessions/list.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570&bid=19774994)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/controllers/public/sessions/list.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570&bid=19774994)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/controllers/public/sessions/list.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570&bid=19774994)

File: app/controllers/public/sessions/list.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  toggleTrack() {
+    this.toggleProperty('isTrackVisible');
+  }
+
+  @computed('model.event.startsAt', 'model.event.endsAt')
+  get allDates() {
+    let arr = new Array();
+    let difference = (this.model.event.endsAt).diff(this.model.event.startsAt, 'days');
+    for (let i = 0; i <= Math.abs(difference); i++) {
+    	arr.push(moment(this.model.event.startsAt).add(i,'days').toISOString());
```
Comment (niranjan94) [inline app/controllers/public/sessions/list.js:16]: Codacy found an issue: ['moment' is not defined.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- app/controllers/public/sessions/list.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570&bid=19774994)

File: app/controllers/public/sessions/list.js (hunk 1 above)
Comment (niranjan94) [inline app/controllers/public/sessions/list.js:16]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570)

File: app/controllers/public/sessions/list.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  isTrackVisible = false;
+  @action
+  toggleTrack() {
+    this.toggleProperty('isTrackVisible');
+  }
+
+  @computed('model.event.startsAt', 'model.event.endsAt')
+  get allDates() {
+    let arr = new Array();
+    let difference = (this.model.event.endsAt).diff(this.model.event.startsAt, 'days');
```
Comment (niranjan94) [inline app/controllers/public/sessions/list.js:14]: Codacy found an issue: ['difference' is never reassigned. Use 'const' instead.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570)

File: app/controllers/public/sessions/list.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+export default class extends Controller {
+  isTrackVisible = false;
+  @action
+  toggleTrack() {
+    this.toggleProperty('isTrackVisible');
+  }
+
+  @computed('model.event.startsAt', 'model.event.endsAt')
+  get allDates() {
+    let arr = new Array();
```
Comment (niranjan94) [inline app/controllers/public/sessions/list.js:13]: Codacy found an issue: ['arr' is never reassigned. Use 'const' instead.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570)

File: app/controllers/public/sessions/list.js (hunk 1 above)
Comment (niranjan94) [inline app/controllers/public/sessions/list.js:16]: Codacy found an issue: [A space is required after ','.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570)

**Comment by maze-runnar** [comment]
video of working - https://youtu.be/gdotFtocMjc

File: app/controllers/public/sessions/list.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  toggleTrack() {
+    this.toggleProperty('isTrackVisible');
+  }
+
+  @computed('model.event.startsAt', 'model.event.endsAt')
+  get allDates() {
+    const arr = new Array();
+    const difference = (this.model.event.endsAt).diff(this.model.event.startsAt, 'days');
+    for (let i = 0; i <= Math.abs(difference); i++) {
+    	arr.push(moment(this.model.event.startsAt).add(i, 'days').toISOString());
```
Comment (niranjan94) [inline app/controllers/public/sessions/list.js:16]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/controllers/public/sessions/list.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570&bid=19774994)

File: app/controllers/public/sessions/list.js (hunk 4 above)
Comment (niranjan94) [inline app/controllers/public/sessions/list.js:16]: Codacy found an issue: ['moment' is not defined.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/controllers/public/sessions/list.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570&bid=19774994)

File: app/controllers/public/sessions/list.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
```
Comment (niranjan94) [inline app/controllers/public/sessions/list.js:17]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570)

**Review by mariobehling — CHANGES_REQUESTED** [review]
Please implement the design more closely to the website generator. Do not use the track color for the entire area. Compare https://summit.fossasia.org/event/schedule.html

Screenshot from 2020-08-20 10-00-12

**Comment by mariobehling** [comment]
In the screenshots, I dont see if all details are present. Here is the list what should definitely be there:
* Session Title
* Track
* Type (workshop, talk etc.)
* Speaker Name and links
* Time and Date
* Room
* Description

**Comment by maze-runnar** [comment]
> In the screenshots, I dont see if all details are present. Here is the list what should definitely be there:
> 
> * Session Title
> * Track
> * Type (workshop, talk etc.)
> * Speaker Name and links
> * Time and Date
> * Room
> * Description

@mariobehling  I have updated the changes . please take a look - https://youtu.be/gdotFtocMjc,

**Comment by maze-runnar** [comment]
@iamareebjamal @mariobehling  please take a look at these changes - 
Screenshot from 2020-08-20 17-01-06
Screenshot from 2020-08-20 17-01-00

**Comment by iamareebjamal** [comment]
Don't show time in card. Try to mimick https://summit.fossasia.org/event/schedule.html as much as you can. Remove the outer card as well

Expanded session should show these details: https://github.com/fossasia/open-event-frontend/pull/4796#issuecomment-677447189

Not in the old UI. New UI is set to replace the old UI. So all the information should be available in the new UI itself including the date switcher as well

**Comment by maze-runnar** [comment]
@iamareebjamal i have made it similar to https://summit.fossasia.org/event/schedule.html .
Screenshot from 2020-08-20 18-48-50

Screenshot from 2020-08-20 18-48-41

**Comment by maze-runnar** [comment]
Screenshot from 2020-08-20 18-48-50

**Comment by iamareebjamal** [comment]
Looks good. Please remove the time from card, only session box should be in a card. I'll review the implementation soon

**Comment by maze-runnar** [comment]
> Looks good. Please remove the time from card, only session box should be in a card. I'll review the implementation soon

ok

**Comment by iamareebjamal** [comment]
Also, use `isLight` utility function to change the text color in the card

**Comment by maze-runnar** [comment]
> Also, use `isLight` utility function to change the text color in the card

which color?

**Comment by maze-runnar** [comment]
@iamareebjamal please take a look -
Screenshot from 2020-08-20 19-29-19
Screenshot from 2020-08-20 19-29-16

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/controllers/public/sessions/list.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570&bid=19774994)

File: app/controllers/public/sessions/list.js (hunk 5 above)
Comment (niranjan94) [inline app/controllers/public/sessions/list.js:17]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6034570)

**Comment by iamareebjamal** [comment]
Session Title and subtitle should be of same color. And there is still card nesting. The color card is sitting inside a white card

**Comment by maze-runnar** [comment]
> Session Title and subtitle should be of the same color. And there is still card nesting. The color card is sitting inside a white card

No there is no card nesting, this is default accordion.

**Comment by maze-runnar** [comment]
> Session Title and subtitle 

they should be white ?

**Comment by iamareebjamal** [comment]
Yes

**Comment by mariobehling** [comment]
I think the implementation here is more closely related to this issue https://github.com/fossasia/open-event-frontend/issues/4807

It seems you are implementing the tracks view page.

What is the difference between the different schedule pages? 
1. Schedule Page in calendar view (in future different view should be switchable)
1.a) Tracks Page - all sessions are sorted in list view according to date, time and track
1.b) Rooms Page - all sessions are sorted in list view according to date, time and room
2. Session Page - this page shows all sessions in alphabetical order (including accepted and confirmed ones that are not yet scheduled)

**Comment by iamareebjamal** [comment]
Great work @maze-runnar. I'm going to take your UI and logic to implement #4808 as it is high priority and to be fixed by tomorrow. And then later, we can use the same components to implement track and schedule view.

Thanks for doing the hard work for me

**Comment by iamareebjamal** [comment]
UI changed now

**Pull request closed without merging**